src/Bicep.Core/Registry/Oci/OciManifest.cs

```diff
@@ -8,26 +8,34 @@ namespace Bicep.Core.Registry.Oci
 {
     public class OciManifest
     {
-        public OciManifest(int schemaVersion, string? artifactType, OciDescriptor config, IEnumerable<OciDescriptor> layers, IDictionary<string, string>? annotations = null)
+        public OciManifest(int schemaVersion, string? mediaType, string? artifactType, OciDescriptor config, IEnumerable<OciDescriptor> layers, OciDescriptor? subject = null, IDictionary<string, string>? annotations = null)
         {
-            this.Annotations = annotations?.ToImmutableDictionary() ?? ImmutableDictionary<string, string>.Empty;
+            this.Annotations = (annotations?.Count > 0) ? annotations.ToImmutableDictionary() : null;
             this.SchemaVersion = schemaVersion;
             this.ArtifactType = artifactType;
             this.Config = config;
+            this.Subject = subject;
             this.Layers = layers.ToImmutableArray();
+            MediaType = mediaType;
         }
 
         public int SchemaVersion { get; }
 
+        public string? MediaType { get; }
         public string? ArtifactType { get; }
 
         public OciDescriptor Config { get; }
 
         public ImmutableArray<OciDescriptor> Layers { get; }
 
+        /// <summary>
+        /// Reference to a separate manfest that this manifest is being attached to
+        /// </summary>
+        public OciDescriptor? Subject { get; }
+
         /// <summary>
         /// Additional information provided through arbitrary metadata.
         /// </summary>
-        public ImmutableDictionary<string, string> Annotations { get; }
+        public ImmutableDictionary<string, string>? Annotations { get; }
```

Comment:
>Annotations [](http://example.com/codeflow?start=52&length=11)

Can we use customize the serializer we use to not emit the `Annotations` property if it's empty. If we make it nullable, all callers have to deal with it potentially being null.

Reply:
Dealing with possibly null it is quite easy, especially for this class' usage.  I'd rather not add complexity when not needed.